modules/actor/actor-wfrp4e.js

```diff
@@ -1283,125 +1283,155 @@ export default class ActorWfrp4e extends WFRP4eDocumentMixin(Actor)
+    chatData.speaker = { alias: this.name };
+    ChatMessage.create(chatData);
   }
 
+  /**
+   * Finishes disease and handles Lingering symptoms.
+   *
+   * @param {ItemWfrp4e} disease
+   *
+   * @return {Promise<void>}
+   */
   async finishDisease(disease) {
-
     let msg = game.i18n.format("CHAT.DiseaseFinish", { disease: disease.name })
 
-    if (disease.system.symptoms.includes("lingering")) {
+    if (disease.system.symptoms.value.includes("lingering")) {
       let lingering = disease.effects.find(e => e.name.includes("Lingering"))
       if (lingering) {
-        let difficulty = lingering.name.substring(lingering.name.indexOf("(") + 1, lingering.name.indexOf(")")).toLowerCase()
-
-        this.setupSkill(game.i18n.localize("NAME.Endurance"), { difficulty }).then(setupData => this.basicTest(setupData).then(async test => {
-          if (test.result.outcome == "failure") {
-            let negSL = Math.abs(test.result.SL)
-            if (negSL <= 1) {
-              let roll = (await new Roll("1d10").roll()).total
-              msg += game.i18n.format("CHAT.LingeringExtended", { duration: roll })
-            }
-            else if (negSL <= 5) {
-              msg += game.i18n.localize("CHAT.LingeringFestering")
-              fromUuid("Compendium.wfrp4e-core.diseases.kKccDTGzWzSXCBOb").then(disease => {
-                this.createEmbeddedDocuments("Item", [disease.toObject()])
-              })
-            }
-            else if (negSL >= 6) {
-              msg += game.i18n.localize("CHAT.LingeringRot")
-              fromUuid("Compendium.wfrp4e-core.diseases.M8XyRs9DN12XsFTQ").then(disease => {
-                this.createEmbeddedDocuments("Item", [disease.toObject()])
-              })
-            }
+        let difficulty = lingering.name.substring(lingering.name.indexOf("(") + 1, lingering.name.indexOf(")")).toLowerCase();
+
+        let test = await this.setupSkill(game.i18n.localize("NAME.Endurance"), { difficulty });
+        test.roll();
+
+        if (test.result.outcome === "failure") {
+          let negSL = Math.abs(test.result.SL);
+          let lingeringDisease;
+
+          if (negSL <= 1) {
+            let roll = (await new Roll("1d10").roll()).total;
+            msg += game.i18n.format("CHAT.LingeringExtended", { duration: roll });
+          } else if (negSL <= 5) {
+            msg += game.i18n.localize("CHAT.LingeringFestering");
+            lingeringDisease = await fromUuid("Compendium.wfrp4e-core.diseases.kKccDTGzWzSXCBOb");
+          } else if (negSL >= 6) {
+            msg += game.i18n.localize("CHAT.LingeringRot");
+            lingeringDisease = await fromUuid("Compendium.wfrp4e-core.diseases.M8XyRs9DN12XsFTQ");
           }
-        }))
+
+          if (lingeringDisease)
+            await this.createEmbeddedDocuments("Item", [lingeringDisease.toObject()]);
+        }
       }
+    } else {
+      // await this.deleteEmbeddedDocuments("ActiveEffect", [removeEffects]);
```

Comment:
`removeEffects` was not defined at all in the function, so commenting it for now. I realize that these functions will need an additional pass once Diseases and Symptoms are migrated.